app/controllers/discover.js

```diff
@@ -171,6 +171,15 @@ export default Ember.Controller.extend({
             this.set('sortByOptions', copy);
             this.set('page', 1);
             this.loadPage();
+        },
+
+        selectSubjectFilter(subject) {
+            let match = this.get('activeFilters').filter(function(item) {
+                return item.indexOf(subject.text) !== -1;
+            });
+            if (!match.length) {
+                this.get('activeFilters').pushObject('subject:' + subject.text);
```

Comment:
activeFilters holds all filters. It should be an object with ` { type of filter: values }`, but right now it is a list of strings with prefixes. One of my todos after demo is refactor this.